doc/source/data/dataset-tensor-support.rst

```diff
@@ -3,446 +3,198 @@
 ML Tensor Support
 =================
 
-Tables with tensor columns
-~~~~~~~~~~~~~~~~~~~~~~~~~~
+Tensor (multi-dimensional array) data is ubiquitous in ML workloads. However, popular data formats such as Pandas, Parquet, and Arrow don't natively support Tensor data types. To bridge this gap, Datasets provides a unified Tensor data type that can be used to represent and store Tensor data:
 
-Datasets supports tables with fixed-shape tensor columns, where each element in the column is a tensor (n-dimensional array) with the same shape. As an example, this allows you to use Pandas and Ray Datasets to read, write, and manipulate e.g., images. All conversions between Pandas, Arrow, and Parquet, and all application of aggregations/operations to the underlying image ndarrays are taken care of by Ray Datasets.
+* For Pandas, Datasets will transparently convert ``List[np.ndarray]`` columns to and from the :class:`TensorDtype <ray.data.extensions.tensor_extension.TensorDtype>` extension type.
+* For Parquet, the Datasets Arrow extension :class:`ArrowTensorType <ray.data.extensions.tensor_extension.ArrowTensorType>` allows Tensors to be loaded and stored in Parquet format.
+* In addition, single-column Tensor datasets can be created from image and Numpy (.npy) files.
 
-With our Pandas extension type, :class:`TensorDtype <ray.data.extensions.tensor_extension.TensorDtype>`, and extension array, :class:`TensorArray <ray.data.extensions.tensor_extension.TensorArray>`, you can do familiar aggregations and arithmetic, comparison, and logical operations on a DataFrame containing a tensor column and the operations will be applied to the underlying tensors as expected. With our Arrow extension type, :class:`ArrowTensorType <ray.data.extensions.tensor_extension.ArrowTensorType>`, and extension array, :class:`ArrowTensorArray <ray.data.extensions.tensor_extension.ArrowTensorArray>`, you'll be able to import that DataFrame into Ray Datasets and read/write the data from/to the Parquet format.
+Datasets automatically converts between the extension types/arrays above. This means you can just think of "Tensors" as a single first-class data type in Datasets.
 
-Automatic conversion between the Pandas and Arrow extension types/arrays keeps the details under-the-hood, so you only have to worry about casting the column to a tensor column using our Pandas extension type when first ingesting the table into a ``Dataset``, whether from storage or in-memory. All table operations downstream from that cast should work automatically.
+Creating Tensor Datasets
+------------------------
 
-Single-column tensor datasets
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This section shows how to create single and multi-column Tensor datasets.
```

Comment:
Hmm I'm not sure if casting this as "single-column" vs "multi-column" is the best way to frame this, since we're now trying to hide the former as an implementation detail for a "collection of tensors" concept. We haven't successfully hidden it completely, the returned schema still shows that it's a single-column table, but I think that's still the goal.

The way I've been thinking about it is that we have a "tensor dataset" concept that represents a "collection of tensors", and it uses a single-tensor-column table under-the-hood as it's concrete data representation, but all APIs to interact with the data should provide a tensor (i.e. NumPy) API. This is addressing the pure-tensor (e.g. only images) use case.

I also think that the more common multi-column use case is a mix of column types (e.g. N feature columns, one or more of which are tensor columns, and a non-tensor label column), not multiple tensor columns. So this single-tensor-column vs multi-tensor-column framing doesn't seem to line up with the underlying use cases.

I think that the ideal framing would be "tensor datasets", consisting of a single-tensor-column table that presents a "collection of tensors" concept and API to the user, and "tensors in tabular datasets" or "tensors in tables", where a table can contain one or more tensor columns along with other column types. What do you think?

Reply:
Yeah, I don't really buy it. In the long run, I really think tensor dataset should go away as a terminology, at best informally used.

For example, once Arrow/Pandas support it officially--- we can probably retire the term altogether, and it will just be a normal feature (i.e., we'll have Datasets, plus our special `__value__` shorthand and that's it).